electives/18741.md

```diff
@@ -0,0 +1,52 @@
+# 18-441/18-741: Computer Networks
+
+| Category | Difficulty |
+|:--       | :-:        |
+| Quiz     | 5          |
+| Projects | 7          |
+| Exams    | 7          |
+
+This class is a great introduction to Computer Networks covering all the 
+layers of the network stack with the physical and data-link layer covered in the first half
+of the course and network, transport and application layer covered in the second half
+of the course. The course does not assume background with socket programming and dealing
+with highly concurrent programs, however experience with the same through other systems courses
+like 18-213 should be great help. The course is quite comprehensive in providing a whirlwind tour
+of networks and helps in building good intuition and experience in network programming. This class
+used to traditionally be offered in the Spring but has now been shifted to the Fall.
+
+## Language Used
+All projects in the course are mandatory to be completed in Python, which is well known for its
+great in-built data structures, parsing and library support. It allows for students inexperienced with systems programming before to comfortably work on the projects in the given timeframe without getting bogged down by syntactic or arcane memory related issues. 
+Prior experience in dealing with python data structures such as lists, dictionaries and familiarity with the rich string manipulation and parsing functions as well as numerical libraries such as numpy should greatly aid in completing the projects.
+
+## Lectures
+
+The lectures are an integral part of the course with key concepts explained in class through examples
+along with the discussions and questions asked by students during class. The quizzes and exams rely heavily on material discussed in class hence attending lectures should help immensely. The first few
+the lectures cover the physical layer involving details around modulation, bit encoding, link bandwidth and noise immunity. The next few lectures deep dive into the data-link layer covering concepts around flow control, medium access control, framing and error detection and correction. The second half of the course focuses heaily on the network and transport layer and talks about routing protocols and algorithms, addressing and naming in the Internet. The lectures also cover the two most well known transport protocols including UDP and TCP, their semantics, techniques for congestion control. The last few lectures dive into the application layer protocol and techniques that are transforming the network such as content delivery networks, software defined networks, 5G and other interesting research topics. The recommended textbook for this course is Computer Networks: A Systems Approach by Bruce S. Davie and Larry L. Peterson, online free versions of which are available on the Internet as well as at CMU Libraries. The library also has hard copies of the book available as well.   
```

Comment:
would be nice to add a link to the textbook if you can

Reply:
Added the link as well